app/src/org/commcare/activities/MenuBase.java

```diff
@@ -0,0 +1,79 @@
+package org.commcare.activities;
+
+import android.content.Intent;
+import android.os.Bundle;
+import android.view.View;
+import android.widget.AdapterView;
+
+import org.commcare.CommCareApplication;
+import org.commcare.session.SessionFrame;
+import org.commcare.suite.model.Entry;
+import org.commcare.suite.model.Menu;
+
+public abstract class MenuBase
+        extends SaveSessionCommCareActivity
+        implements AdapterView.OnItemClickListener {
+
+    private boolean isRootModuleMenu;
+    protected String menuId;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        menuId = getIntent().getStringExtra(SessionFrame.STATE_COMMAND_ID);
+
+        if (menuId == null) {
+            menuId = Menu.ROOT_MENU_ID;
+            isRootModuleMenu = true;
+        }
+    }
+
+    /**
+     * Stores the path of selected form and finishes.
+     */
+    @Override
+    public void onItemClick(AdapterView listView, View view, int position, long id) {
+        String commandId;
+        Object value = listView.getAdapter().getItem(position);
+
+        // if value is null, probably it means that we clicked on the header view, so we just ignore it
+        if (value == null) {
+            return;
+        }
```

Comment:
This null check wasn't present in `MenuGrid` but I don't see any issue including it in the common super class.